core/src/main/java/com/linecorp/armeria/client/retry/RetryStrategyBuilder.java

```diff
@@ -0,0 +1,191 @@
+
+    RetryStrategyBuilder() {}
+
+    @Override
+    public RetryStrategyBindingBuilder onIdempotentMethods() {
+        return newBindingBuilder().onIdempotentMethods();
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onMethods(HttpMethod... methods) {
+        return newBindingBuilder().onMethods(methods);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onMethods(Iterable<HttpMethod> methods) {
+        return newBindingBuilder().onMethods(methods);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onStatusClass(HttpStatusClass... statusClasses) {
+        return newBindingBuilder().onStatusClass(statusClasses);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onStatusClass(Iterable<HttpStatusClass> statusClasses) {
+        return newBindingBuilder().onStatusClass(statusClasses);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onServerErrorStatus() {
+        return newBindingBuilder().onServerErrorStatus();
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onStatus(HttpStatus... statuses) {
+        return newBindingBuilder().onStatus(statuses);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onStatus(Iterable<HttpStatus> statuses) {
+        return newBindingBuilder().onStatus(statuses);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onStatus(Predicate<? super HttpStatus> statusFilter) {
+        return newBindingBuilder().onStatus(statusFilter);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onException(Class<? extends Throwable> exception) {
+        return newBindingBuilder().onException(exception);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onException(Predicate<? super Throwable> exceptionFilter) {
+        return newBindingBuilder().onException(exceptionFilter);
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onException() {
+        return newBindingBuilder().onException();
+    }
+
+    @Override
+    public RetryStrategyBindingBuilder onUnProcessed() {
+        return newBindingBuilder().onUnProcessed();
+    }
+
+    /**
+     * Adds a {@link RetryStrategy}.
+     */
+    public RetryStrategyBuilder add(RetryStrategy retryStrategy) {
+        retryStrategiesBuilder.add(requireNonNull(retryStrategy, "retryStrategy"));
+        return this;
+    }
+
+    /**
+     * Adds a {@link RetryRule}.
+     *
+     * <p><pre>{@code
+     * RetryStrategy.builder()
+     *              .on(RetryRule.onStatus(HttpStatus.SERVICE_UNAVAILABLE)
+     *                           .onException(ex -> ex instanceof ClosedSessionException)
+     *                           .onMethod(HttpMethod.GET)
+     *                           .thenBackOff(myBackoff))
+     *              .build();
+     * }</pre>
+     */
+    RetryStrategyBuilder on(RetryRule retryRule) {
+        return add(build(requireNonNull(retryRule, "retryRule")));
+    }
```

Comment:
For simplicity, how about:

- Removing the `on*()` methods in this class
  - We will be able to merge `AbstractRetryStretegyBindingBuilder` into `RetryRuleBuilder`.
- Renaming this method to something like `rule()`

..?

Reply:
That's good. I removed and simplified a lot.